src/three-components/SmoothControls.ts

```diff
@@ -154,6 +139,44 @@ export interface ChangeEvent extends Event {
   source: string,
 }
 
+/**
+ * The Damper class is a generic second-order critically damped system that does
+ * one linear step of the desired length of time. The only parameter is
+ * DECAY_MILLISECONDS, which should be adjustable: TODO(#580). This common
+ * parameter makes all states converge at the same rate regardless of scale.
+ * xNormalization is a number to provide the rough scale of x, such that
+ * NIL_SPEED clamping also happens at roughly the same convergence for all
+ * states.
+ */
+export class Damper {
+  private[$velocity]: number = 0;
+
+  update(
+      x: number, xGoal: number, timeStepMilliseconds: number,
+      xNormalization: number): number {
+    if (timeStepMilliseconds >= DECAY_MILLISECONDS) {
+      // This clamps at the point where a large time step would cause the
+      // discrete step to overshoot.
+      this[$velocity] = 0;
+      return xGoal;
+    }
+    if (timeStepMilliseconds < 0)
+      return x;
+    // Critically damped
+    const acceleration = NATURAL_FREQUENCY * NATURAL_FREQUENCY * (xGoal - x) -
+        2 * NATURAL_FREQUENCY * this[$velocity];
+    this[$velocity] += acceleration * timeStepMilliseconds;
+    if (Math.abs(this[$velocity]) < NIL_SPEED * xNormalization &&
+        acceleration * (xGoal - x) <= 0) {
+      // This ensures the controls settle and stop calling this function instead
+      // of asymptotically approaching their goal.
+      this[$velocity] = 0;
+      return xGoal;
+    } else
+      return x + this[$velocity] * timeStepMilliseconds;
```

Comment:
This should be enforced by linting in some way, but conventionally we always express `if/else` clauses as blocks. Please wrap this clause and the others above in braces `{ }`

Filed https://github.com/GoogleWebComponents/model-viewer/issues/608 to make improvements here

Reply:
Not sure why this doesn't show as outdated, but I did fix this, along with any other unbracketed if statements I could find. I'm used to the Google C++ style guide where the brackets are omitted.